csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs

```diff
@@ -100,6 +101,38 @@ public static SessionOptions MakeSessionOptionWithTensorrtProvider(int deviceId
             }
         }
 
+        /// <summary>
+        /// A helper method to construct a SessionOptions object for TensorRT execution provider.
+        /// Use only if CUDA/TensorRT are installed and you have the onnxruntime package specific to this Execution Provider.
+        /// </summary>
+        /// <param name="trtProviderOptions">TensorRT EP provider options</param>
+        /// <returns>A SessionsOptions() object configured for execution on provider options</returns>
+        public static SessionOptions MakeSessionOptionWithTensorrtProvider(OrtTensorRTProviderOptions trtProviderOptions)
+        {
+            CheckTensorrtExecutionProviderDLLs();
+            SessionOptions options = new SessionOptions();
+            try
+            {
+                // get device id for configuring CUDA EP
+                int deviceId;
+                string optionsStr;
+                var dict = new Dictionary<string, string>();
+                optionsStr = trtProviderOptions.GetOptions();
+                ProviderOptionsValueHelper.StringToDict(optionsStr, dict);
+                deviceId = Int32.Parse(dict["device_id"]);
```

Comment:
>dict["device_id" [](http://example.com/codeflow?start=39&length=16)

What happens the entry is not present?

Reply:
After code refactor, this part is removed.